### ISSUE TITLE
How to safely reduce the number of training length for an epoch

### ISSUE THREAD

**Issue by dingjibang** [facebookresearch/demucs#307]
## ❓ Questions

I noticed that the training size of an epoch is defined by num_examples of [wav.py](https://github.com/facebookresearch/demucs/blob/main/demucs/wav.py#L142)

But as the training data increases, an epoch training take >12 hours, which is unbearable, how to safely shorten this num_example to the original 1/10, and can basically guarantee 10 new epoch = 1 old epoch, I want to shorten this time to see the results and SDR scores of each epoch faster

Suppose there are 100 pairs of files, for example, 

epoch1 only trains the 1st to 10th pairs of files...
epoch2 trains the 11th to 20th pairs of files....
... 

This is feasible in code, but compared to the full full file training, Will it affect the results of the model?

**Comment by dingjibang** [comment]
After I read the code, I found the **max_batches** parameter, which solved my problem perfectly (thanks for the robust code design), just in case I also looked at the logic of wav.__getitem__, which is based on DataLoader (distrib.loader method) and use the shuffled index

Sorry to bother you, it's my own problem
